src/backend/web/routes/posts.js

```diff
@@ -3,10 +3,11 @@ const express = require('express');
 const Post = require('../../data/post');
 const { getPosts, getPostsCount } = require('../../utils/storage');
 const { logger } = require('../../utils/logger');
+const { validatePostsQuery } = require('../validation');
 
 const posts = express.Router();
 
-posts.get('/', async (req, res) => {
+posts.get('/', validatePostsQuery, async (req, res) => {
```

Comment:
For consistency, let's always have middleware be accessible via a function call (always returns a new function) vs. a property (which can be altered a runtime):

```js
posts.get('/', validatePostsQuery(), async (req, res) => {
```